### PR TITLE
NVTX Range Transform

### LINKED ISSUE #2754: NVTX range push/pop transform
**Is your feature request related to a problem? Please describe.**
NVTX ranges are being defined by push and pop mechanism, currently the nvtx transforms need to be define be separate push and pop transforms (#2713) but it would be useful to have `RangePushPop`  as well, as a more convenient way to annotate a single transform.

### PR DESCRIPTION
**Pull request by drbeh**
Fixes #2754 

### Description
This PR adds Range transform which accept a transform as an input and pushes an NVTX range before that transform, and pops it afterwards. 

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_nvtx_transform.py [hunk 1]
```diff
@@ -116,33 +119,92 @@ def test_nvtx_transfroms_array(self, input):
...
                 RangePopD(),
-                MarkD("Mark: Transform Ends!"),
+                MarkD("Mark: Transforms (p=1) End!"),
+            ]
+        )
+        output = transforms(input)
+        self.assertIsInstance(output["image"], torch.Tensor)
+        np.testing.assert_array_equal(input["image"], Flip()(output["image"].numpy()))
+
+    # ---------------------
```
Comment (Nic-Ma) [inline tests/test_nvtx_transform.py:156]: Please remove these debug comments.
Thanks.

File: tests/test_nvtx_transform.py [hunk 2]
```diff
@@ -116,33 +119,92 @@ def test_nvtx_transfroms_array(self, input):
...
+            ]
+        )
+        output = transforms(input)
+        self.assertIsInstance(output, torch.Tensor)
+        np.testing.assert_array_equal(input, output)
+        # with prob == 1.0
+        transforms = Compose(
+            [
+                RandMark("Mark: Transforms (p=1) Start!"),
+                Range(RandFlip(prob=1.0)),
```
Comment (Nic-Ma) [inline tests/test_nvtx_transform.py:177]: Could you please also add test cases for `RandRange`?

Thanks.

Comment (drbeh) [reply tests/test_nvtx_transform.py]: Thanks for the comment. I added `RandRange` and `RandRangeD`.

**Review by Nic-Ma — APPROVED** [review]
Looks good to me.
@wyli what do you think?

Thanks.

**Comment by drbeh** [comment]
@wyli, if you don't have any comment on this, I'll merge the PR. Please let me know. Thanks.

**Comment by wyli** [comment]
thanks @drbeh, I have a quick question, what will happen if we have the same `type(transform).__name__` msg for different nvtx transforms?

**Comment by drbeh** [comment]
> thanks @drbeh, I have a quick question, what will happen if we have the same `type(transform).__name__` msg for different nvtx transforms?

If we are profiling the same transform that is used multiple times in the compose, like `ToTensor`, currently, the only way to distinguish them is to assign different name to it. For instance: `Range(ToTensor, "ToTensor_1")` instead of `Range(ToTensor)`.

**Pull request merged**